### PR TITLE
[e2e][azure] Make internalStaticIP flexible

### PR DESCRIPTION
**Pull request by lzhecheng**
Now, internalStaticIP is hard-coded to "10.240.11.11". Such IP works
for aks-engine cluster but not for CAPZ ones (node-subnet 10.1.0.0/16)

Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

#### What type of PR is this?

/kind failing-test
#### What this PR does / why we need it:
Now, internalStaticIP is hard-coded to "10.240.11.11". Such IP works
for aks-engine cluster but not for CAPZ ones (node-subnet 10.1.0.0/16)
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/network/loadbalancer.go
Comment (aojea) [inline test/e2e/network/loadbalancer.go:674]: now that I see it, this test has a lot of assumptions :/
L590
```
		isInternalEndpoint := func(lbIngress *v1.LoadBalancerIngress) bool {
			ingressEndpoint := e2eservice.GetIngressPoint(lbIngress)
			// Needs update for providers using hostname as endpoint.
			return strings.HasPrefix(ingressEndpoint, "10.")
		}
```

File: test/e2e/network/loadbalancer.go
Comment (aojea) [inline test/e2e/network/loadbalancer.go:696]: there are already helpers in `test/e2e/framework/node/resource.go` , please double check but I think that you can 

```
node,err := e2enode.GetRandomReadySchedulableNode()
internalIP, err := e2enode.GetInternalIP(node)
```

File: test/e2e/network/loadbalancer.go
Comment (aojea) [inline test/e2e/network/loadbalancer.go:672]: ditto, let's see if we can get of hardcoded values and make the test really portable

**Comment by aojea** [comment]
These tests are no "Conformance" :smile:

File: test/e2e/network/loadbalancer.go
Comment (lzhecheng) [reply test/e2e/network/loadbalancer.go:676]: Indeed. Updated them all.

File: test/e2e/network/loadbalancer.go
Comment (lzhecheng) [reply test/e2e/network/loadbalancer.go]: Yes, but actually for azure, the control-plane doesn't have the same subnet as workers. So I created a new method to pick a worker node.

File: test/e2e/network/loadbalancer.go
Comment (lzhecheng) [reply test/e2e/network/loadbalancer.go]: Yes, but I think hard-code it to something like `xx.xx.11.11` is enough for an e2e test?
Do you mean avoiding `11.11` as well? Then, do we need to check all other resources to avoid a same IP?

**Comment by lzhecheng** [comment]
> These tests are no "Conformance" 😄

You're right. Title changed. I used to call every test in k/k as `conformance` but it may not be the truth...

**Review by feiskyer — APPROVED** [review]
/lgtm

File: test/e2e/network/loadbalancer.go
Comment (aojea) [reply test/e2e/network/loadbalancer.go]: that-'s ok ... I vaguely remember some method  doing exactly the same , but I can't find it ...  :smile:

File: test/e2e/framework/node/resource.go
Comment (aojea) [inline test/e2e/framework/node/resource.go:380]: nit, SubnetPrefix assumin is a /16 :upside_down_face:

**Comment by aojea** [comment]
/approve
Thanks
/retest

File: test/e2e/framework/node/resource.go
Comment (khenidak) [reply test/e2e/framework/node/resource.go:380]: right, I don't see how we can get subnet from an IP or even all nodes' IPs. is cloud provider vendor-ed into e2e? 
/hold until this is fully understood.

Comment (khenidak) [reply test/e2e/framework/node/resource.go:380]: Or does the test always run with a /16 subnet? if so can we please update the test doc?

Comment (lzhecheng) [reply test/e2e/framework/node/resource.go:380]: I updated the test doc. Yes, this test assumes a /16 subnet just like it used to do.

**Review by feiskyer — COMMENTED** [review]
/lgtm

File: test/e2e/framework/node/resource.go
Comment (aojea) [reply test/e2e/framework/node/resource.go:380]: @khenidak this area has a lot of tech debt, just bear in mind this test was much worse before this PR
https://github.com/kubernetes/kubernetes/pull/109413/files#r847224114

I suggest to merge now and iterate on the subnet assumption later ... I defer to you to unhold

Comment (lzhecheng) [reply test/e2e/framework/node/resource.go:380]: @khenidak could you take a look at @aojea 's suggestion? Now v1.24 is released.

**Pull request merged**
